pandas/core/arrays/base.py

```diff
@@ -46,11 +51,18 @@ def try_cast_to_ea(cls_or_instance, obj, dtype=None):
     -------
     ExtensionArray or obj
     """
-    try:
-        result = cls_or_instance._from_sequence(obj, dtype=dtype)
-    except Exception:
-        # We can't predict what downstream EA constructors may raise
-        result = obj
+    if not (
```

Comment:
so i *really* don't like this. What  are the possibilities for cls_or_instance here?

Reply:
Actually @jorisvandenbossche concluded that those conditions should be removed from there and we should fix this in a more meaningful manner. [see this comment](https://github.com/pandas-dev/pandas/pull/30282#issuecomment-575686004)